File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        c = Counter("".join(words))
        l = len(words)
        return all(map(lambda x: x%l == 0,c.values()))
        # for v in c.values():
        #     if v % l != 0:
        #         return False
        

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.php
class Solution {

    /**
     * @param String[] $words
     * @return Boolean
     */
    function makeEqual($words) {
        $l = count($words);
        $counter = array();
        foreach($words as $w) {
            for($i = 0;$i < strlen($w);$i++) {
                $c = $w[$i];
                $counter[$c] += 1;
            }
        }
        print_r($counter);
        foreach($counter as $v) {
            if($v % $l !== 0) {
                return false;
            }
        }
        return true;
    }
}